src/QuadForm/Quad/ZLattices.jl

```diff
@@ -1479,17 +1479,77 @@ julia> R = root_lattice_recognition(L)
 ```
 """
 function root_lattice_recognition(L::ZLat)
-  return _connected_components(root_sublattice(L))
+  irr = irreducible_components(root_sublattice(L))
+  return Tuple{Symbol, Int}[ADE_type(gram_matrix(i)) for i in irr], irr
 end
 
-function _connected_components(L::ZLat)
+@doc Markdown.doc"""
+    irreducible_components(L::ZLat)
+
+Return the irreducible components ``L_i`` of the positive definite lattice ``L``.
+
+This yields a maximal orthogonal splitting of `L` as
+```math
+L = \bigoplus_i L_i.
+```
+The upper bound is used for the short vector computation.
```

Comment:
Hm, which upper bound?

Reply:
Right, I removed it from the interface. Thx.